modules/post/windows/recon/host_info.rb

```diff
@@ -0,0 +1,149 @@
+  def recon_hostname
+    print_good("Hostname is #{get_hostname}")
+    get_hostname
+  end
+
+  # Obtains the IP address the session uses to connect to the machine and
+  # learns its interface and mac address
+  def recon_address
+    interface = nil
+    mac = nil
+    if session.type == 'meterpreter'
+      session.net.config.each_interface do |iface|
+        if iface.addrs.include?(session.session_host)
+          interface = iface.mac_name.strip
+          mac = iface.mac_addr.unpack('H*').first.gsub(/([\da-f]{2})/, '\1:').delete_suffix(':')
+        end
+      end
+    end
+    if interface.blank? || mac.blank?
+      if session.type == 'powershell'
+        # If we have powershell, then we do all of our work in powershell
+        interface = cmd_exec("(Get-NetAdapter|Where-Object{$_.Name -eq (Get-NetIPAddress|Where-Object{$_.IPAddress -eq '#{session.session_host}'}).InterfaceAlias}).InterfaceDescription").strip
+        mac = cmd_exec("(Get-NetAdapter|Where-Object{$_.Name -eq (Get-NetIPAddress|Where-Object{$_.IPAddress -eq '#{session.session_host}'}).InterfaceAlias}).MacAddress").gsub('-', ':').strip
+      # elsif have_powershell?
+      #   interface = psh_exec("(Get-NetAdapter|Where-Object{$_.Name -eq (Get-NetIPAddress|Where-Object{$_.IPAddress -eq '#{session.session_host}'}).InterfaceAlias}).InterfaceDescription")
+      #   mac = psh_exec("(Get-NetAdapter|Where-Object{$_.Name -eq (Get-NetIPAddress|Where-Object{$_.IPAddress -eq '#{session.session_host}'}).InterfaceAlias}).MacAddress").gsub('-', ':')
+      else
+        # Since theres no powershell, attempt to claim it using netsh
+        interface = cmd_exec('netsh interface ip show ipaddresses').scan(/Interface \d+: ([^\r\n]+)\r\n\r\nAddr Type  DAD State   Valid Life Pref. Life Address\r\n---------  ----------- ---------- ---------- ------------------------\r\n\S+\s+\S+\s+\S+\s+\S+\s+(\d{,3}\.\d{,3}\.\d{,3}\.\d{,3})\r\n/)
+
+        # if the interface query gave anything, then check which one pertains to our IP address
+        interface = interface.find { |x| x[1] == session.session_host }[0] if interface.any?
+
+        if !interface.empty?
+          # Now that we have our interface time to get our mac address
+          mac_list = CSV.parse(cmd_exec('getmac /v /fo csv'))
+          mac = mac_list.find { |row| row[0] == interface }[2].gsub('-', ':')
+        end
+      end
+    end
+    print_good("The session is running on address #{session.session_host} (#{mac}) on interface #{interface}")
+    report_host(host: session.session_host, mac: mac) if active_db? && !mac.blank?
+  end
+
+  # Obtains information about the system architecture
+  def recon_architecture
+    if session.type == 'meterpreter'
+      host_arch = sysinfo['Architecture']
+    elsif session.type == 'powershell'
+      host_arch = case cmd_exec('(Get-ComputerInfo).OSArchitecture').strip
+                  when '64-bit'
+                    'x64'
+                  when '32-bit'
+                    'x32'
```

Comment:
```suggestion
                    'x86'
```

Reply:
How does the underlying framework detect architecture? Part of why its there is to give more information for other modules to use such as knowing what payloads can be used on the target. Will `x86` be fully recognized as the same as `x32` for modules and payloads?